src/frontend/Komponenter/Behandling/Vurdering/EndreVurderingComponent.tsx

```diff
@@ -62,43 +61,38 @@ const EndreVurderingComponent: FC<{
 
     const oppdaterBegrunnelse = (vurderinger: Vurdering[], index: number, nyttSvar: Vurdering) => {
         const { begrunnelse } = nyttSvar;
+        const svarsalternativ: Svarsalternativ | undefined = hentSvarsalternativ(regler, nyttSvar);
+        if (!svarsalternativ) {
+            return;
+        }
 
         const oppdaterteSvar = oppdaterSvarIListe(nyttSvar, vurderinger, true);
 
-        const svarsalternativ: Svarsalternativ | undefined = hentSvarsalternativ(regler, nyttSvar);
-        if (svarsalternativ) {
-            const nesteStegId = svarsalternativ?.regelId;
-            const maybeLeggTilNesteNodIVilkårsvar = leggTilNesteIdHvis(
-                nesteStegId,
-                oppdaterteSvar,
-                () =>
-                    nesteStegId !== 'SLUTT_NODE' &&
-                    begrunnelseErPåkrevdOgUtfyllt(svarsalternativ, begrunnelse) &&
-                    !vurderinger.find((v) => v.regelId === nesteStegId)
-            );
-            oppdaterVilkårsvar(index, maybeLeggTilNesteNodIVilkårsvar);
-        }
+        const maybeLeggTilNesteNodIVilkårsvar = leggTilNesteIdHvis(
+            svarsalternativ.regelId,
+            oppdaterteSvar,
+            () => begrunnelseErPåkrevdOgUtfyllt(svarsalternativ, begrunnelse)
+        );
+        oppdaterVilkårsvar(index, maybeLeggTilNesteNodIVilkårsvar);
         settIkkePersistertKomponent(vurdering.id);
     };
 
     const oppdaterSvar = (vurderinger: Vurdering[], index: number, nyttSvar: Vurdering) => {
-        const oppdaterteSvar = oppdaterSvarIListe(nyttSvar, vurderinger);
         const svarsalternativer: Svarsalternativ | undefined = hentSvarsalternativ(
             regler,
             nyttSvar
         );
-
-        if (svarsalternativer) {
-            const maybeLeggTilNesteNodIVilkårsvar = leggTilNesteIdHvis(
-                svarsalternativer.regelId,
-                oppdaterteSvar,
-                () =>
-                    svarsalternativer.regelId !== 'SLUTT_NODE' &&
-                    svarsalternativer.begrunnelseType !== BegrunnelseRegel.PÅKREVD
-            );
-
-            oppdaterVilkårsvar(index, maybeLeggTilNesteNodIVilkårsvar);
+        if (!svarsalternativer) {
+            return;
         }
+        const oppdaterteSvar = oppdaterSvarIListe(nyttSvar, vurderinger);
```

Comment:
Nå husker jeg ikke logikken her, men er det trygt å kjøre return på linjen over FØR vi oppdaterer svaret? 😬

Reply:
Er i prinsipp det samme som var fra før, der vi ikke gjorde noe hvis man ikke hadde svarsalternativer :)
Det skulle implicitt betyde at vi ikke har en regelmapping for svaret (eller ikke har satt svarId da)

Reply:
(dvs return blir bare tydligere og får "exit fast")